### PR TITLE
Fetch items by given uri

### PR DESCRIPTION
**Pull request by annando**
From now on we can simply enter an url of a remote post in the search field and it will be fetched.

This is currently a draft since I'm unsure about the different classes. I want to add this stuff in some way to ```mod/search.php``` but there is some ```src/Module/BaseSearchModule.php``` as well.

My function will be short. Simply calling ```$id = Item::fetchByLink($uri);``` then jumping to the display page. But I'm unsure if I can place it directly into ```mod/search.php``` or if our new class structure suggests doing it differently from this.

### CODE REVIEW

**Comment by nupplaphil** [comment]
That's my fault, I started moving search to module but first I was too late for the release and now I fixed the config problems.

I'll go on moving the search module after you're finished :)

So for now just add the logic the `mod/search.php` way ;)

**Comment by annando** [comment]
@nupplaphil the PR is now ready to test.

File: src/Model/Item.php [hunk 1]
```diff
@@ -3575,4 +3576,67 @@ public static function isForumPost(array $item, array $owner = [])
...
+	 *
+	 * @param string $uri
+	 * @param integer $uid
+	 *
+	 * @return integer item id
+	 */
+	public static function searchByLink($uri, $uid = 0)
+	{
+		$ssl_uri = str_replace('http://', 'https://', $uri);
+		$uris = [$uri, $ssl_uri, Strings::normaliseLink($uri)];
```
[the remark this answers is not in the document]
Comment (annando) [reply src/Model/Item.php]: Not really. The normalize function also does cut a possibly trailing ```/```. So you could enter ```http://domain.tld/post/guid/``` which would be changed to ```https://domain.tld/post/guid/``` and ```http://domain.tld/post/guid```.

[the remark this answers is not in the document]
Comment (MrPetovan) [reply src/Model/Item.php]: Fair enough, I wasn't aware of that quirk.

**Comment by MrPetovan** [comment]
It works like a charm with a top-level post and a comment, this is awesome!

**Comment by annando** [comment]
Sadly it doesn't work that great with Pleroma, since the visible URL can't be used. But it does - of course - work with Friendica links as well.

In the future I will have a look at Diaspora as well.

**Pull request merged**